### PR TITLE
[WIP] Feature/ch11530/read smart contract

### PR DESCRIPTION
**Pull request by mahsa-bytt**
this PR is about smart contract query implementation in SDK-java. (this function is implemeneted in JS-SDK).
I checkout this branch from https://github.com/quantnetwork/overledger-sdk-java/tree/feature/ch11529/invokesmartcontract, because some of the objects that I need for this task, is implemented in that branch(invoke-smart-contract task).
so it should be merged after that.
one important point is that ethereum-connector doesn't support ARRAY type, and we can't send array_type in our request.( currently in JS-SDK if we send array type, we get error).
The implementation follows the exact flow of JS-SDK to be same as each other.

### CODE REVIEW

File: overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumUtil.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                    return null;
+                }).collect(Collectors.toUnmodifiableList());
+    }
+
+        public static List<Object> getValues(List<ContractArgument> inputs) {
+        return inputs.stream().map(contractArgument -> {
+            ContractInputTypeOptions type = contractArgument.getType();
+            String typeAsString = type.label;
+            try {
+                if (type.name().toLowerCase().contains(ARRAY_TYPE)) {
```
Comment (richardbahari) [inline overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumUtil.java:47]: could you move the IF and else into a couple of methods?

File: overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumUtil.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                    if (typeAsString.contains(BOOL_TYPE))
+                        return Boolean.parseBoolean(value);
+                    else if (typeAsString.contains(ADDRESS_TYPE))
+                        return new Address(value);
+                    else if (typeAsString.contains(INT_TYPE))
+                        return new BigInteger(value);
+                    else if (typeAsString.contains(BYTES_TYPE))
+                        return new byte[]{Byte.parseByte(value)};
+                }
+                return null;
```
Comment (richardbahari) [inline overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumUtil.java:67]: is return null still required here?

File: overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumUtil.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            String typeAsString = type.label;
+            try {
+                if (type.label.contains(INT_TYPE) || type.label.contains(UINT_TYPE))
+                    solidityInputType = addIntegerLengthToInput(typeAsString, input.getSelectedIntegerLength());
+                else if (typeAsString.contains(BYTES_TYPE))
+                    solidityInputType = addByteOptionsToInput(typeAsString, input.getSelectedBytesLength());
+                else
+                    solidityInputType = typeAsString;
+
+                if (type.name().toLowerCase().contains(ARRAY_TYPE)) {
```
Comment (richardbahari) [inline overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumUtil.java:113]: should this be moved before the IF type.label.contains? May save the trouble of processing if type.label and else if.

**Review by richardbahari — COMMENTED** [review]
a few comments

File: overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumAccount.java [hunk 4]
```diff
@@ -27,15 +29,16 @@
 @Slf4j
 public class EthereumAccount implements Account {
 
-    private static final BigInteger DEFAULT_MAIN_GAS_LIMIT = new BigInteger("8000000");
-    private static final BigInteger DEFAULT_TEST_GAS_LIMIT = new BigInteger("4712388");
+    private static final BigInteger DEFAULT_MAIN_GAS_LIMIT = BigInteger.valueOf(8000000);
+    private static final BigInteger DEFAULT_TEST_GAS_LIMIT = BigInteger.valueOf(4712388);
```
Comment (lukerQuant) [inline overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumAccount.java:33]: This is from previous developer right? This is out of date and the node would check for this anyway I believe

Comment (lukerQuant) [reply overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumAccount.java:33]: Current mainnet gas limit is around 11m?? -> but yeah we don't need these limits hard coded in

Comment (mahsa-bytt) [reply overledger-sdk-ethereum/src/main/java/network/quant/ethereum/EthereumAccount.java:33]: yes, this is from previous one.
I just changed new BigInteger to BigInteger.valueOf() (sonarLint hint)

**Review by kashif-quant — APPROVED** [review]

**Pull request closed without merging**